### ISSUE TITLE
Looking for 4.0.0 Beta Testers

### ISSUE THREAD

**Issue by tsightler** [tsightler/ring-mqtt#100]
Hi all!

Over the last couple of weeks I've been working hard on refactoring the code to simplify it and make it easier to maintain so that I can add new features and keep it working with less overall maintenance going forward.  I'm nearing release of the 4.0.0 version but I'd really like to get some testing from a few others if there's anyone out there that like to play with the "cutting edge".  Below is a list of features for 4.0.0:

* Support for alarm status
* Full support for Home Assistant device registry (multiple entities to one device)
* Info sensor for each device to monitor extended attributes of devices (things like comm status, ac connection, battery level and charging state, alarm details, etc, exact details vary per device).  Battery status is charted for most devices.
* Battery/ac status for security keypad monitoring
* Panic buttons as part of Alarm Control Panel device which can also be used to monitor type of alarm (optionally enabled)

When used with the new 4.0.0 addon, there are even more features:
* Dramatically simplified configuration (MQTT configuration is now handled automatically and completely removed from the config).  Literally install, use the web UI to generate a token, and your off
* Web UI is fully integrated into Home Assistant via Ingress
* Automatic monitoring of Home Assistant restarts (no config needed)
* The addon now uses pre-built Docker images from Docker Hub instead of building dynamically

I'm still missing a few features before I hit the 4.0.0 milestone:
* Info sensor for cameras (Wifi/battery status, etc)
* Info support for base station
* Siren support for alarm (manual siren trigger)

Unfortunately, there are a few breaking changes:
* I decided to change the MQTT topic hierarchy **AGAIN**.  I think the new topic levels are much more elegant and allow for a very clear hierarchy.  For Home Assistant users this should be transparently handled by the auto discovery process as the unique entity ids are the same so this mostly impacts those using tools like Node-Red.
* Battery/tamper state is now surfaced via the the new "info" sensor and is no longer available as JSON attributes on the main sensor.  If you were using automations based on the old json attributes they will need to be updated to use the new info sensor data.
* Because of changes with the addon config, upgrades may require manually editing the config the first time, it may be easier to remove the old add and re-install.

If anyone out there is willing to test the latest 4.0.0-beta3, it is upgraded to the dev branch, and, if you are using the addon, you can use the [development addon](https://github.com/tsightler/ring-mqtt-ha-addon-dev).

**Comment by xrolfex** [comment]
I'll give it a go, switching now.

**Comment by stephywephy88** [comment]
The Dev add-on is working well for me. I renamed some of my Ring devices because of overlap ("Frontdoor" device = frontdoor doorbell? frontdoor contact sensor? frontdoor lock?) to make it easier for me to quickly see the device type. It was time for a device name cleanup anyway in the Ring app. I'm having z-wave connection trouble with one of my Gen 1 sensors (just ordered a second extender, per Ring analysis) and it's the only device that shows up as "link quality: unknown."

I'm still trying to figure out a entry-delay automation. I want my entry lights to turn red if someone opens the door when the alarm is in entry-delay (in my case, home or away modes). I see my alarmState change to entry-delay in a Lovelace card, but in DevTools the alarm state changes to "pending" during entry-delay mode. I changed the automation's conditional state from "entry-delay" to "pending" in my automation and that didn't trigger it either. I was following these state names and will keep playing around.

burglar-alarm
entry-delay
fire-alarm
co-alarm
panic
user-verified-burglar-alarm
user-verified-co-or-fire-alarm
burglar-accelerated-alarm

I did need to Open the Web UI four times to regenerate my token over the course of config'n until I got things settled.

Thank you for your ongoing work on this - I will donate to a charity as you preferred!

**Comment by tsightler** [comment]
Thanks for testing!  I'm glad it is working well for you.  I'm nearly ready to push 4.0.0-beta4 which adds the last of the planned 4.0.0 features.

Regarding the entry delay, you can use the alarm_control_panel or the info sensor data for your automation.  The MQTT alarm control panel in home assistant only supports a small subset of status data.  The addon previously supported only disarmed, armed_home and armed_away, but now also supports the pending and triggered status.

The pending status in the alarm control panel is effectively the same as entry-delay and is probably the easiest way to do something specifically on entry.  The only option once the alarm goes to this mode is it can be disarmed, or will eventually go triggered after the entry delay expires.

However, I know some people wanted even finer grained automation for alarm types, so that's why the information sensor provides the full detail of the type of alarm so they can take different actions based on, for example, fire alarm vs burglar alarm (for example, with burglar you may want to also trigger other alarms on cameras, etc, while that's pretty useless for fire).  With the info sensor you can get extremely fine grained with your automation, but probably isn't needed for simple entry delay automation.

**Comment by christopherdopp** [comment]
I installed the beta version of 4.0.0 and everything went very smoothly. Once I remembered how to add a custom git repo correctly (ie. not in HACS but rather the Add-on store in the Supervisor),  it installed and started up correctly. I only had to open the Web UI once to login and enter my verification code. I stopped my instance of ring-mqtt that I had running on a different Raspberry Pi on my network before starting the beta version and didn't encounter any issues. My sensors all kept working after switching to 4.0.0 and I noticed the new info sensors and the data they contain. Haven't used it for very long, but it seems to be working properly without any problems that I saw. Thanks for all your hard work on this.

**Comment by stephywephy88** [comment]
Thanks - that helped me unravel when to use entry-delay versus pending but I just could not get it to trigger. When I stopped my regular add-on and added the dev add-on, all my Ring sensors in HomeKit stopped responding. So I went back to the regular add-on, which brought the devices back in Homekit and I made the delay-entry automation using Homekit.  Something in my yaml automation must have been wrong. I'll be happy to test 4.0.0-beta4 too when you give the word.
